web-services/query/src/main/java/datawave/webservice/query/runner/RunningQuery.java

```diff
@@ -332,11 +331,8 @@ public ResultsPage next() throws Exception {
                 }
             }
         }
-        if (resultList.isEmpty()) {
-            return new ResultsPage();
-        } else {
-            return new ResultsPage(resultList, ((hitPageByteTrigger || hitPageTimeTrigger) ? ResultsPage.Status.PARTIAL : ResultsPage.Status.COMPLETE));
-        }
+        ResultsPage.Status status = (currentPageCount < maxPageSize) ? ResultsPage.Status.PARTIAL : ResultsPage.Status.COMPLETE;
```

Comment:
We are so close here, however a subtlety was missed.  If the page being returned is the last page to be returned, then this is not a "partial page".  So if we have hit the max results, or it.hasNext() is false, then lets return COMPLETE in those cases.